### PR TITLE
scripts: Print file causing error for syscalls

### PR DESCRIPTION
**Pull request by yperess**
I had an issue with a file I was using that caused the `read()` call to fail, but I wasn't able to find which file until I made this change. Hopefully this will make it easier for the next person.

### CODE REVIEW

File: scripts/build/parse_syscalls.py [hunk 1]
```diff
@@ -77,7 +77,11 @@ def analyze_headers(multiple_directories):
                     continue
 
                 with open(path, "r", encoding="utf-8") as fp:
-                    contents = fp.read()
+                    try:
+                        contents = fp.read()
+                    except Exception:
+                        sys.stderr.write("Error decoding %s\n" % path)
```
Comment (tejlmand) [inline scripts/build/parse_syscalls.py:83]: you commit message states:
`that caused the `open` call to fail`
but this `try-except` is on the `read()`, not the `open()` ?

Generally, errors on `open()` will print the file is question, so could you please provide more details on the error / exception seen.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Not sure what problem this PR fixes.

Commit message says one thing, code does something else.

Please also read the contribution guideline wrt. commit messages:
https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-body

The commit message:
```
I had an issue with a file I was writing that caused the `open` call
to fail, but I wasn't able to find which file until I made this change.
```
doesn't help understand the root of the problem.

Please include some details on exactly which exception was thrown.
All details don't need to be in commit message, but in the PR some details on exact error would be nice.

File: scripts/build/parse_syscalls.py (hunk 1 above)
Comment (yperess) [reply scripts/build/parse_syscalls.py:83]: Sorry, you're right, I wrote the commit message days after making the change and didn't write it correctly.

**Review by nordicjm — CHANGES_REQUESTED** [review]
Nothing has changed in this commit with regards to @tejlmand's requested changes. Also please fix the commit message to have a summary of the changes, not how you encountered an issue

**Comment by yperess** [comment]
> Nothing has changed in this commit with regards to @tejlmand's requested changes. Also please fix the commit message to have a summary of the changes, not how you encountered an issue

I'm sorry, I just have forgotten to push. I'll update shortly.

**Review by nordicjm — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Pull request merged**